lib/evm_database_ops.rb

```diff
@@ -28,47 +28,49 @@ def self.database_size(opts)
   end
 
   def self.backup(db_opts, connect_opts = {})
-    # db_opts:
-    #   :dbname => 'vmdb_production',
-    #   :username => 'root',
-    #   :local_file => "/tmp/backup_1",      - Backup locally to the file specified
-
-    # connect_opts:
-    #   :uri => "smb://dev005.manageiq.com/share1",
-    #   :username => 'samba_one',
-    #   :password => 'Zug-drep5s',
-    #   :remote_file_name => "backup_1",     - Provide a base file name for the uploaded file
+    uri = run_db_opt_for_new_uri(db_opts, connect_opts) do |database_opts|
+      PostgresAdmin.backup(database_opts)
+    end
+    _log.info("[#{db_opts[:dbname]}] database has been backed up to file: [#{uri}]")
+    uri
+  end
 
+  # db_opts:
+  #   :dbname => 'vmdb_production',
+  #   :username => 'root',
+  #   :local_file => "/tmp/backup_1",      - Backup locally to the file specified
+
+  # connect_opts:
+  #   :uri => "smb://dev005.manageiq.com/share1",
+  #   :username => 'samba_one',
+  #   :password => 'Zug-drep5s',
+  #   :remote_file_name => "backup_1",     - Provide a base file name for the uploaded file
+  private_class_method def self.run_db_opt_for_new_uri(db_opts, connect_opts = {})
```

Comment:
This method name is not great. I would prefer we refactor this into more pieces to make the individual operations more obvious. For example, `with_mount_session` and `check_free_space`